### ISSUE TITLE
Bug: FPS cap during login (background cap/vsync), results in lower then possible FPS after login to game.

### ISSUE THREAD

**Issue by IngeniousDox** [acomminos/wine-pba#19]
Was on Varimatras Mythic, got a portal to Dalaran, took it, ended up in old Dalaran. I walked towards the AH there, I get a lagspike. Then I'm suddenly on 12 fps. CPU usage is still 280%, GPU usages is 35%. My fps never goes back up to 60 (vsync capped).

I couldn't reproduce it, but I figured I would document and make a proper bug issue. Perhaps someone else will stumble upon a way to trigger this bug consistently.

This feels related to bug that I have during testing where I sometimes I have lower fps then expected after logging in. You would expect no matter how you log in or after a stutter, that your fps will stabilize on max fps. It simply does not.

**Comment by acomminos** [comment]
If you're still using revision 68de8e9b3f26e68bc6d64f353e0954ddab2f7590, you're very likely to be encountering the issue that c565623357098b1c946b4ef6c7768f123630e27f fixes (running out of free list entries randomly, which would cause a synchronous path to be taken).

On that note, I can't imagine how c565623357098b1c946b4ef6c7768f123630e27f might actually be causing a performance regression. I'll try and reproduce locally.

**Comment by IngeniousDox** [comment]
I'm indeed still using https://github.com/acomminos/wine-pba/commit/68de8e9b3f26e68bc6d64f353e0954ddab2f7590, and perhaps this is 2 bugs indeed, the free list problem and the login issue. I'll update issue to only have the login issue.

I have been able to reproduce the same lower fps after login with https://github.com/acomminos/wine-pba/commit/c565623357098b1c946b4ef6c7768f123630e27f when I was testing it yesterday. I have also just repeated it, steps:
- Have background fps capped at 30 fps, no cap on foreground. 
- Start game
- Load into character, location: Astravar Harbor, Suramar, on top of building. 53.75, 70.25. (Though it happens in other locations aswell)
- See 50 fps instead of expected 60 fps (vsync / triplebuffer)

Here is the glxosd output: [glxosd_benchmark_login_ends_in_50_fps.csv.tar.gz](https://github.com/acomminos/wine-pba/files/1777290/glxosd_benchmark_login_ends_in_50_fps.csv.tar.gz)

Now your other thought, perhaps https://github.com/acomminos/wine-pba/commit/c565623357098b1c946b4ef6c7768f123630e27f doesn't have a regression, it might have to do with this login bug, since I did my testing in skyhold just after restarting the game. But like said, it was only a feeling, and it could have had a cause that had nothing to with pba. (I'm actually thinking it could be stutters from adding stuff to the nvidia shader cache could effect it aswell. It gets cleared when you use a new version of Wine). So my suggestion is to just shelve my feelings about "smoothness", ignore it completely, and spend your time on coding away at other stuff.
